### PR TITLE
Do not let EventsTimetable schedule past events if catchup=False

### LINKED ISSUE #33948: EventsTimetable schedules past events even if catchup=False
### Apache Airflow version

2.7.0

### What happened

Relates to investigation of #30327

When a DAG with a timetable with `schedule=EventsTimetable()` and `catchup=False` is defined, all planned events are executed even if they are in the past.

Note: If this is not a bug but a feature, the docs should mention it that the `EventsTimetable` ignores the `catchup` flag. But docs do not tell about this.

### What you think should happen instead

With `catchup=False` I would have expected no past events are scheduled or maximum the one matching the last occurrence.

### How to reproduce

Setup a test DAG with an event time table like the following example and enable scheduling. Wait a moment and all past event dates are executed.

```
with DAG(
    dag_id="after_workday_events_regression",
    start_date=pendulum.datetime(2023, 8, 1, tz="UTC"),
    catchup=False,
    schedule=EventsTimetable(
        event_dates=[
            pendulum.datetime(2023, 8, 1, 1, 0, tz="America/Chicago"),
            pendulum.datetime(2023, 8, 2, 1, 0, tz="America/Chicago"),
            pendulum.datetime(2023, 8, 3, 1, 0, tz="America/Chicago"),
        ],
        description="My Team's Baseball Games",
        restrict_to_events=False,
    ),
    params={"test": 123},
):
    @task
    def test(ti: TaskInstance=None):
        print(ti.execution_date)
    test()
```

### Operating System

Ubuntu 20.04 / Breeze Dev setup in Py 3.8 Container

### Versions of Apache Airflow Providers

not relevant.

### Deployment

Other

### Deployment details

Started and tested with latest main branch and breeze.

### Anything else

I tried to find the root cause but was not able to locate it. Suspect it is rooted in `scheduler_job_runner.py`.
I'd supply a fix but was not able to fund the root cause. Expert knowledge is needed probably.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by avkirilishin**
closes: https://github.com/apache/airflow/issues/33948

Fixed the bug and added some more tests.

### CODE REVIEW

File: airflow/timetables/events.py
Comment (Taragolis) [inline airflow/timetables/events.py]: I would recommend to use `airflow.utils.timezone.utcnow` for this one instead of direct use `pendulum.DateTime.utcnow()`.

We might potentially get rid of `pendulum` in the future, and this might help to replace this easier

**Review by jscheffl — APPROVED** [review]
Did a regressio with the example included in the bug report and this PR fixed it.

**Comment by jscheffl** [comment]
@uranusjr As you are tagged as Codeowner, can you make another review before merge? Alternatively @Taragolis please make a second pass and then OK to merge?

File: airflow/timetables/events.py
Comment (uranusjr) [inline airflow/timetables/events.py]:
```suggestion
        if not self.event_dates:
            return None
        for next_event in self.event_dates:
            if earliest and next_event < earliest:
                continue
            elif last_automated_data_interval and next_event <= last_automated_data_interval.end:
                continue
            break
```

Not exactly sure but this seems to be logically equivalent?

**Comment by uranusjr** [comment]
Looks right to me in general, just a minor idea on implementation. I also pushed a commit to improve typing slightly.

**Review by eladkal — APPROVED** [review]

**Pull request merged**
